lib/factory.rb

```diff
@@ -0,0 +1,82 @@
+require 'pry'
+
+class Factory
+  class << self
+    def new(*args, &block)
+      const_set(args.shift.capitalize, new(*args, &block)) if args.first.is_a? String
```

Comment:
what if I receive empty string?
And u should check that ur const don`t defined early